### DISCUSSION TITLE
New device detection for an existing user in Supabase

### DISCUSSION THREAD

**Discussion by fbele** [supabase/supabase#21393] [Questions] [answered]
Hi there,

It's important for my use case, that I have a different (new) device detection, something like for example Google, or PayPal has.
I want to know if Supabase already has some out-of-the-box mechanism to handle such situation, or do I need to implement it myself?

Basically I'd like to have the following scenario:
1. A new user signs up in the app and his current device is being saved in the DB (Supabase). As long as he is logging in from now on only using this device, there is nothing else to consider.
2. Later on, this same user logs in the app with a new device. At this point the Supabase needs to be able to detect that it's a new device that the user is using and he needs to confirm that he is logging in with the new device, using a link in an email or a confirm push notifications (the channel is not relevant at this point).

What would be the best way to implement such flow?

Thanks for any help in advance!

**Comment by GaryAustin1** [comment]
The closest Supabase has built in is this pro feature to limit a user to one session.  https://supabase.com/docs/guides/auth/sessions#limiting-session-lifetime-and-number-of-allowed-sessions-per-user

**Reply by fbele** [reply]
Hi Gary,

Thank you for your swift reply and support! I thought so, that such a solution needs to be implemented manually.

One off-topic question to this thread, but I don't want to open a new Question for it - does the `auth.users` table, or generally all the tables in `auth` schema, have Row-Level Security turned on by default? And if no, do you think it's necessary to do so? Or is there another security mechanism, so that turning on the RLS on these tables is not necessary?

Thanks.

**Reply by GaryAustin1** [reply, marked as answer]
The auth schema is not accessible from the API at all.  Also none of the postgres roles the API uses have permission to access that schema.
